modules/encoders/cmd/base64.rb

```diff
@@ -0,0 +1,72 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Encoder
+  Rank = GoodRanking
+
+  BASE64_BYTES = [
+    'A'.ord...'Z'.ord,
+    'a'.ord...'z'.ord,
+    '0'.ord...'9'.ord
+  ].map(&:to_a).flatten + '+/='.bytes
+
+  def initialize
+    super(
+      'Name' => 'Base64 Command Encoder',
+      'Description' => %q{
+        This encoder uses base64 encoding to avoid bad characters.
+      },
+      'Author' => 'Spencer McIntyre',
+      'Arch' => ARCH_CMD,
+      'Platform' => %w[linux unix],
+      'EncoderType' => Msf::Encoder::Type::CmdPosixBase64)
+
+    register_advanced_options(
+      [
+        OptString.new('Base64Decoder', [ false, 'The binary to use for base64 decoding', '', %w[base64 openssl] ])
+      ],
+      self.class
+    )
+  end
+
+  #
+  # Encodes the payload
+  #
+  def encode_block(state, buf)
+    return buf if state.badchars.empty?
```

Comment:
Shouldn't `buf` be returned as well if it doesn't contain any bytes present in `state.badchars`?